lib/charms/operator_libs_linux/v0/grub.py

```diff
@@ -1,24 +1,67 @@
-"""TODO: Add a proper docstring here.
+# Copyright 2023 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
 
-This is a placeholder docstring for this charm library. Docstrings are
-presented on Charmhub and updated whenever you push a new version of the
-library.
+"""Simple library for managing Linux kernel configuration via GRUB.
 
-Complete documentation about creating and documenting libraries can be found
-in the SDK docs at https://juju.is/docs/sdk/libraries.
+This library is only used for setting additional parameters that will be stored in the
+"/etc/default/grub.d/95-juju-charm.cfg" config file and not for editing other
+configuration files. It's intended to be used in charms to help configure the machine.
 
-See `charmcraft publish-lib` and `charmcraft fetch-lib` for details of how to
-share and consume charm libraries. They serve to enhance collaboration
-between charmers. Use a charmer's libraries for classes that handle
-integration with their charm.
+Configurations for individual charms will be stored in "/etc/default/grub.d/90-juju-<charm>",
+but these configurations will only have informational value as all configurations will be merged
+to "/etc/default/grub.d/95-juju-charm.cfg".
 
-Bear in mind that new revisions of the different major API versions (v0, v1,
-v2 etc) are maintained independently.  You can continue to update v0 and v1
-after you have pushed v3.
+Example of use:
 
-Markdown is supported, following the CommonMark specification.
+```python
+class UbuntuCharm(CharmBase):
+    def __init__(self, *args):
+        ...
+        self.framework.observe(self.on.install, self._on_install)
+        self.framework.observe(self.on.update_status, self._on_update_status)
+        self.framework.observe(self.on.remove, self._on_remove)
+        self.grub = grub.GrubConfig(self.meta.name)
+        log.debug("found keys %s in GRUB config file", self.grub.keys())
+
+    def _on_install(self, _):
+        try:
+            self.grub.update(
+                {"GRUB_CMDLINE_LINUX_DEFAULT": '"$GRUB_CMDLINE_LINUX_DEFAULT hugepagesz=1G"'}
```

Comment:
Presumably this example shouldn't have the quoting, given that we're passing the key-value strings? Similar below in `_on_update_status`.

Reply:
I'll updated it. Thanks